tests/e2e/cluster_downgrade_test.go

```diff
@@ -140,30 +163,36 @@ func stopEtcd(t *testing.T, ep e2e.EtcdProcess) {
 }
 
 func validateVersion(t *testing.T, cfg *e2e.EtcdProcessClusterConfig, member e2e.EtcdProcess, expect version.Versions) {
-	// Two separate calls to expect as it doesn't support multiple matches on the same line
-	var err error
-	testutils.ExecuteWithTimeout(t, 20*time.Second, func() {
+	testutils.ExecuteWithTimeout(t, 30*time.Second, func() {
 		for {
-			if expect.Server != "" {
-				err = e2e.SpawnWithExpects(e2e.CURLPrefixArgs(cfg, member, "GET", e2e.CURLReq{Endpoint: "/version"}), nil, `"etcdserver":"`+expect.Server)
+			err := func() error {
```

Comment:
Please remove creating anonymous function. There is no need for it if we immidietly call it.

Reply:
The `func` is used to unify the `time.Sleep` statement. If not, there are several `time.Sleep` if err != nil, like

```
if expect.Cluster != result.Cluster {
    time.Sleep(time.Second)
    continue
}
if expect.Server != result.Server {
    time.Sleep(time.Second)
    continue
} 
...
```

Just wonder that there is any reason to prevent this usage? :joy: Thanks!

Reply:
I would say it obfuscates code. It's better to be explicit than implicit. There is nothing wrong with repetition.

Reply:
Better alternative would be to change anonymous function to normal function that takes `expect` and `result` versions.